src/Session/SessionStorage.php

```diff
@@ -353,9 +353,10 @@ protected function collectGarbage()
             return;
         }
         $divisor = $this->options->getInt('gc_divisor');
+        $probabilityPercentage = 100 * $probability / $divisor;
+        $chance = mt_rand(1, 100);
 
-        $rand = mt_rand(0, $divisor);
-        if ($rand < $probability) {
+        if ($chance <= $probabilityPercentage) {
```

Comment:
Hmm… OK, but previously it would/could never be true.

`$rand` will always be an integer, and by default `$probability` is 1 … so 1 is never going to be less than 1.

Seems this is one to break both our brains :smile: